src/core/analyzer.ts

```diff
@@ -77,6 +78,17 @@ export class Analyzer {
     }
   }
 
+  /**
+   * Returns the best analyzer for the given directory.
+   *
+   * Currently this just creates a simple analyzer with a FS loader rooted
+   * at the given directory, but in the future it may take configuration from
+   * files including polymer.json or similar.
+   */
+  static forDirectory(dirname: string): Analyzer {
```

Comment:
```js
const  analyzer = Analyzer.forDirectory(str);`
```

nit: This method name is a little off gramatically, or at least I've never seen us use this grammar for a static factory method. Just by looking, I'm unsure if I need to do `Analyzer.forDirectory()` vs. `new Analyzer.forDirectory()` (types help, at least, but JS consumers don't get that). I'd prefer to see `Analyzer.createForDirectory()`, which is explicitly a factory method.

It sounds like all 5 PRs might be based on this, so I'm conscious of how hard this might be to change, but this stood out to me.

Reply:
+1, this is better.